File: Hard/regular_expression_matching.php
<?php

class Solution {

    /**
     * @param String $s
     * @param String $p
     * @return Boolean
     */
    function isMatch($s, $p) {
        return preg_match("/^$p$/", $s);
    }
}

$solution = new Solution;
print_r($solution->isMatch("aa", "a")."\n"); //False
print_r($solution->isMatch("aa", "a*")."\n"); //True
print_r($solution->isMatch("ab", ".*")."\n"); //True

File: Hard/regular_expression_matching.py
import re

class Solution(object):
    def isMatch(self, s, p):
        """
        :type s: str
        :type p: str
        :rtype: bool
        """
        
        if len(s) == 0 or len(s) > 20 or len(p) == 0 or len(p) > 20:
            return False
        
        return bool(re.search("^"+p+"$", s))
        
print(Solution().isMatch("aa", "a")) #False
print(Solution().isMatch("aa", "a*")) #True
print(Solution().isMatch("ab", ".*")) #True